flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/Expression.scala

```diff
@@ -54,6 +54,12 @@ abstract class Expression extends TreeNode[Expression] {
       s"${this.getClass.getName} cannot be transformed to RexNode"
     )
 
+  /**
+    * To support literals on left side of binary expressions, i.e. 12.toExpr % 'a
+    * @return
+    */
+  def toExpr: Expression = this
+
```

Comment:
This should not be put here but in `expressionDsl.scala`.
Btw. I recognized that there is already a function that does exactly what we want. It is called `expr` in `expressionDsl.scala`. You can rename it.